### DISCUSSION TITLE
More variants in the Disclosure component

### DISCUSSION THREAD

**Discussion by marektonik** [twilio-labs/paste#3345] [Ideas] [answered]
Hello there! I've noticed that the current Disclosure component is relatively simple. Could you please consider adding an icon, avatar, or a custom slot before the heading? This enhancement would add more visual appeal and flexibility to the component. Thank you!

<img width="732" alt="Zrzut ekranu 2023-07-24 o 15 39 59" src="https://github.com/twilio-labs/paste/assets/140410476/32f5b2e7-5c2f-44f0-a2fe-ab54b149044c">

**Comment by serifluous** [comment, marked as answer]
Hey @marektonik ! Great suggestion. I'm assuming you're asking about the Figma component. We've added it to our backlog.

**Reply by marektonik** [reply]
@serifluous that's right, I'm asking about the Figma component :) Thanks!

**Reply by marektonik** [reply]
@serifluous we also have a case where additional text to display on the right side would be potentially helpful for users, here is an example. I know we can't do it right now in Figma, but is there an option to compose such disclosure heading in dev?

<img width="808" alt="Zrzut ekranu 2023-09-7 o 14 33 46" src="https://github.com/twilio-labs/paste/assets/140410476/66065fb6-7a0b-40e5-90f8-6715bd838210">

**Reply by serifluous** [reply]
Yep, it's doable! See the documentation here: https://paste.twilio.design/components/disclosure#composition-notes

I also just published the update to the Disclosure Heading in Figma to include slots.
